tests/apollo/util/test_base.py

```diff
@@ -51,22 +51,26 @@ async def wrapper(*args, **kwargs):
 
     return decorator
 
-def retry_test(max_retries: int):
+def repeat_test(max_repeats: int, break_on_first_failure: bool, break_on_first_success: bool):
     """
     Runs a test multiple times until a run succeeds
     """
-
+    assert not (break_on_first_failure and break_on_first_success), \
+        "both flags break_on_first_failure and break_on_first_success cannot be enable at the same time!"
     def decorator(async_fn):
         @wraps(async_fn)
         async def wrapper(*args, **kwargs):
-            for i in range(1, max_retries + 1):
+            for i in range(1, max_repeats + 1):
                 try:
+                    print(f"Running iteration {i}/{max_repeats}, (break_on_first_failure={break_on_first_failure},"
+                          f" break_on_first_success={break_on_first_success})")
                     await async_fn(*args, **kwargs)
-                    break
+                    if break_on_first_success:
+                        break
                 except Exception as e:
-                    log.log_message(message_type='Test attempt failed', run=i, max_retries=max_retries)
-                    if i == max_retries:
+                    log.log_message(message_type='Test attempt failed', run=i, max_repeats=max_repeats)
+                    if i == max_repeats or break_on_first_failure:
                         raise e
-
+                    print(f"Exception caught: {e}, continuing to next iteration (break_on_first_failure is False)..")
```

Comment:
Maybe just add the new parameters to the log message?
you can print the logs to stdout by setting the APOLLO_LOG_STDOUT environment variable (also works with the makefile).